src/main/java/io/cryostat/reports/MemoryCachingReportsService.java

```diff
@@ -108,4 +110,20 @@ public Uni<Map<String, AnalysisResult>> reportFor(ActiveRecording recording) {
     public Uni<Map<String, AnalysisResult>> reportFor(String jvmId, String filename) {
         return reportFor(jvmId, filename, r -> true);
     }
+
+    @Override
+    public boolean keyExists(ActiveRecording recording) {
+        String key = ReportsService.key(recording);
+        return (quarkusCache || memoryCache)
+                && !Objects.isNull(activeCache.as(CaffeineCache.class).getIfPresent(key))
+                && !delegate.keyExists(recording);
+    }
+
+    @Override
+    public boolean keyExists(String jvmId, String filename) {
+        String key = recordingHelper.archivedRecordingKey(jvmId, filename);
+        return (quarkusCache || memoryCache)
+                && !Objects.isNull(archivedCache.as(CaffeineCache.class).getIfPresent(key))
```

Comment:
Maybe `!Objects.isNull(archivedCache.as(CaffeineCache.class).getIfPresent(key))` could be more clearly expressed as `archivedCache.as(CaffeineCache.class).keySet().contains(key)`